content/en/blog/_posts/2019-08-28-cloud-native-java-controller-sdk.md

```diff
@@ -0,0 +1,102 @@
+libraries over decades, even though Golang is getting more and more fancy and 
+popular for developers. Additionally, nowadays developers are able to develop 
+their aggregated-apiservers over SQL-storage and Java has way better support on SQLs.
+
+
+## How to use?
+
+Take maven project as example, adding the following dependencies into your dependencies: 
+
+```xml
+<dependency>
+    <groupId>io.kubernetes</groupId>
+    <artifactId>client-java-extended</artifactId>
+    <version>6.0.1</version>
+</dependency>
+```
+
+Then we can make use of the provided builder libraries to write your own controller. 
+For example, the following one is a simple controller prints out node information 
+on watch notification, see complete example [here](https://github.com/kubernetes-client/java/blob/master/examples/src/main/java/io/kubernetes/client/examples/ControllerExample.java):
+
+```java
+...
+    Reconciler reconciler = new Reconciler() {
+      @Override
+      public Result reconcile(Request request) {
+        V1Node node = nodeLister.get(request.getName());
+        System.out.println("triggered reconciling " + node.getMetadata().getName());
+        return new Result(false);
+      }
+    };
+    Controller controller =
+        ControllerBuilder.defaultBuilder(informerFactory)
+            .watch(
+                (workQueue) -> ControllerBuilder.controllerWatchBuilder(V1Node.class, workQueue).build())
+            .withReconciler(nodeReconciler) // required, set the actual reconciler
+            .withName("node-printing-controller") // optional, set name for controller for logging, thread-tracing
+            .withWorkerCount(4) // optional, set worker thread count
+            .withReadyFunc( nodeInformer::hasSynced) // optional, only starts controller when the cache has synced up
+            .build();
+```
+
+If you notice, the new Java controller framework learnt a lot from the design of 
+[controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) which
+successfully encapsulates the complex components inside controller into several 
+clean interfaces. With the help of Java Generics, we even move on a bit and simply 
+the encapsulation in a better way.
+
+As for more advanced usage, we can wrap multiple controllers into a controller-manager 
+or a leader-electing controller which helps deploying in HA setup. In a word, we can 
+basically find most of the equivalence implementations here from Golang SDK and
+more advanced features are under active development by us.
+
+## Planned for Future 
+
+The community behind the official Kubernetes Java SDK project will be focusing on
+providing more useful utilities for developers who hope to program cloud native
+Java applications to extend Kubernetes. If you are interested in more details, 
+please check out our repo [kubernetes-client/java](https://github.com/kubernetes-client/java).
+Feel free to share also your feedback with us, through Issues or Slack.
```

Comment:
Will readers know what Slack is, and if so which channel in which workspace?

Reply:
added link to the channel